cookbook/import_export/create-custom-step.rst

```diff
@@ -15,36 +15,92 @@ We will begin by creating a NotifyStep with its configuration and a doExecute me
+
+First of all you need to create a new class that will contains the job parameters. This class should implement the following interfaces:
+- ``ConstraintCollectionProviderInterface``
+- ``DefaultValuesProviderInterface``
+- ``FormConfigurationProviderInterface``
 
-.. literalinclude:: ../../src/Acme/Bundle/NotifyConnectorBundle/Handler/CurlHandler.php
+Here we decorate the ``ProductCsvExport`` classes in order to retrieve the proper form export configuration.
+
+.. literalinclude:: ../../src/Acme/Bundle/NotifyConnectorBundle/JobParameters/ProductCsvExportNotify.php
    :language: php
    :linenos:
-   :lines: 1-3,9-
+   :lines: 31-39
 
-Define the Step Element as Service
-----------------------------------
+Then we add the form configuration for our new ``url`` field.
 
-Add your step element to ``steps.yml`` to ensure that it is loaded and processed by ``DependencyInjection/AcmeNotifyConnectorExtension``:
+.. literalinclude:: ../../src/Acme/Bundle/NotifyConnectorBundle/JobParameters/ProductCsvExportNotify.php
+   :language: php
+   :linenos:
+   :lines: 56-69
 
-.. literalinclude:: ../../src/Acme/Bundle/NotifyConnectorBundle/Resources/config/steps.yml
+In order to validate the job parameters we need to define some constraints. Here we want a valid url as input.
+
+.. literalinclude:: ../../src/Acme/Bundle/NotifyConnectorBundle/JobParameters/ProductCsvExportNotify.php
+   :language: php
+   :linenos:
+   :lines: 44-51
+
+We also need to add the default values to our job form.
+
+.. literalinclude:: ../../src/Acme/Bundle/NotifyConnectorBundle/JobParameters/ProductCsvExportNotify.php
+   :language: php
+   :linenos:
+   :lines: 74-79
+
+And don't forget the support method with your job name ``csv_product_export_notify``.
+
+.. literalinclude:: ../../src/Acme/Bundle/NotifyConnectorBundle/JobParameters/ProductCsvExportNotify.php
+   :language: php
+   :linenos:
+   :lines: 85-88
+
+And don't forget to declare this class as a service, with the proper job name as parameter:
+
+.. literalinclude:: ../../src/Acme/Bundle/NotifyConnectorBundle/Resources/config/job_parameters.yml
    :language: yaml
    :linenos:
 
-Configure our new Job
----------------------
+Configure the job profile
+-------------------------
 
-In ``Resources/config/batch_jobs.yml``, we use a first step to export products in csv and we configure the second one to send a notification:
+We need our new form field to be visible on the UI, to do this we need to define our job in the view element as follows:
 
-.. literalinclude:: ../../src/Acme/Bundle/NotifyConnectorBundle/Resources/config/batch_jobs.yml
+.. literalinclude:: ../../src/Acme/Bundle/NotifyConnectorBundle/Resources/config/job_profile.yml
    :language: yaml
    :linenos:
 
-You may have noticed that we use a custom class for the notification step and we define the handler as a step element.
+Translations
+------------
+
+Don't forget to add a translation to our new step:
```

Comment:
same here "don't forget". I am not very comfortable with this as "don't forget" means that readers had the occasion to know this befor,  which is not true. To me you should just write "Add a translation for our brand new step".